### ISSUE TITLE
`Invert If` analyzer should be aware of Is / IsNot complementation.

### ISSUE THREAD

**Issue by AdamSpeight2008** [dotnet/roslyn#51359]
**Brief description:**
Make `Invert If` aware of the compliment pair of `TypeOf ... Is ...` and `TypeOf ... IsNot ...`

**Languages applicable:**
Maybe both, primarily VB.

**Code example that the analyzer should report:**

_Source_
```vb
If TypeOf e Is t Then
  ' Statements 0
Else
  ' Statements 1
End If
```

_Expected_

```vb
If TypeOf e IsNot t Then
  ' Statements 1
Else
  ' Statements 0
End If
```

_Actual_
```vb
If Not TypeOf e Is t Then
  ' Statements 1
Else
  ' Statements 0
End If
```

**Additional information:**

Any more additional information you would like to add.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/master/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by CyrusNajmabadi** [comment]
Agreed. I'm surprised this doesn't work as I thought I made the expression inverter support that.

**Comment by AdamSpeight2008** [comment]
Forgot to include my version details.
```
Microsoft Visual Studio Enterprise 2019 Preview
Version 16.9.0 Preview 3.0
VisualStudio.16.Preview/16.9.0-pre.3.0+30914.41
Microsoft .NET Framework
Version 4.8.03752

Installed Version: Enterprise

C# Tools   3.9.0-3.21056.4+e5eb71d6714749bccf91d9268ece78cd5d4d43ea
Visual Basic Tools   3.9.0-3.21056.4+e5eb71d6714749bccf91d9268ece78cd5d4d43ea
```
